Src/WitsmlExplorer.Frontend/components/Sidebar/GeologyIntervalItem.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useCallback, useContext } from "react";
+import NavigationContext from "../../contexts/navigationContext";
+import NavigationType from "../../contexts/navigationType";
+import OperationContext from "../../contexts/operationContext";
+import OperationType from "../../contexts/operationType";
+import GeologyInterval from "../../models/geologyInterval";
+import { ObjectType } from "../../models/objectType";
+import Well from "../../models/well";
+import Wellbore from "../../models/wellbore";
+import {
+  getContextMenuPosition,
+  preventContextMenuPropagation
+} from "../ContextMenus/ContextMenu";
+import GeologyIntervalContextMenu, {
+  GeologyIntervalContextMenuProps
+} from "../ContextMenus/GeologyIntervalContextMenu";
+import TreeItem from "./TreeItem";
+
+interface GeologyIntervalProps {
+  geology: GeologyInterval;
+  well: Well;
+  wellbore: Wellbore;
+  selected: boolean;
+  nodeId: string;
+  mudloguid: string;
+}
+
+const GeologyIntervalItem = (
+  props: GeologyIntervalProps
+): React.ReactElement => {
+  const { geology: geology, well, wellbore, nodeId, mudloguid } = props;
+  useContext(OperationContext);
+  const { dispatchOperation } = useContext(OperationContext);
+
+  geology["mudloguid"] = mudloguid;
+
+  const { dispatchNavigation, navigationState } = useContext(NavigationContext);
+  const { selectedObject, selectedObjectGroup } = navigationState;
+
+  const isSelected = useCallback(
+    (geology: GeologyInterval) => {
+      return selectedObject &&
+        selectedObjectGroup === ObjectType.geologyInterval &&
+        selectedObject.uid === geology.uid
+        ? true
+        : undefined;
+    },
+    [selectedObject, selectedObjectGroup]
+  );
+
+  const onContextMenu = (
+    event: React.MouseEvent<HTMLLIElement>,
+    GeologyInterval: GeologyInterval[]
+  ) => {
+    preventContextMenuPropagation(event);
+    const contextProps: GeologyIntervalContextMenuProps = {
+      checkedGeologyIntervals: GeologyInterval
+    };
+    const position = getContextMenuPosition(event);
+    dispatchOperation({
+      type: OperationType.DisplayContextMenu,
+      payload: {
+        component: <GeologyIntervalContextMenu {...contextProps} />,
+        position
+      }
+    });
+  };
+
+  return (
+    <TreeItem
+      onContextMenu={(event) => onContextMenu(event, [geology])}
```

Comment:
Got an error while attempting to open the context menu.
![image](https://github.com/equinor/witsml-explorer/assets/48485965/926b5cd4-ac76-4201-9ce0-2dac25c7a1d0)